python/social_bot/teacher.py

```diff
@@ -145,6 +146,25 @@ def add_task_group(self, task_group, weight=1):
         self._task_groups.append(task_group)
         self._weights.append(weight)
 
+    def build_vocab_from_tasks(self):
+        vovab_list = []
+        for g in self._task_groups:
+            for t in g._tasks:
+                vovab_list = vovab_list + t.task_vocab
+        # remove repeated words and convert to dict
+        self._vovab_list = sorted(set(vovab_list),key=vovab_list.index)
+        self.vocab_size = len(self._vovab_list)
+        self._vovab_dict = dict(zip(self._vovab_list,list(range(0,self.vocab_size))))
+
+    def sentence_to_sequence(self, sentence):
+        word_list = sentence.split()
+        sequence = list(map(lambda x : self._vovab_dict[x], word_list))
+        return np.array(sequence)
```

Comment:
length of sequence needs to match the length given in observation space.  We can fill the remaining values using 0 (other word ID start from 1). And we need to check if sequence exceeds the length limit.